osis-core/src/test/java/com/scality/osis/service/impl/ScalityOsisServiceTest.java

```diff
@@ -118,9 +120,16 @@ private void initListTenantMocks() {
 
                         // if filterStartsWith generate customAttributes for all accounts
                         final Map<String, String> customAttributestemp  = new HashMap<>() ;
-                        customAttributestemp.put("cd_tenant_id==" + UUID.randomUUID(), "");
                         data.setCustomAttributes(customAttributestemp);
                         accounts.add(data);
+
+                        if(!StringUtils.isEmpty(request.getFilterKey()) && request.getFilterKey().startsWith(CD_TENANT_ID_PREFIX)) {
+                            // If filter key exists mock only one account in the response with filterKey as customAttributestemp
```

Comment:
left over comment as we are mocking?
Or is it because this PR doesn't cover `filterKey`

Reply:
No this is not a left over comment. We are covering filterKey in this PR. This PR is for QueryTenants where we search for specific tenant using cd_tenant_id via filterKey on listAccounts API.

The idea here is we use the same mock logic for listAccounts to generate the response but will provide only Account in the response if "filterKey" exists in the request and of format "cd_tenant_id==*" i.e., Query Tenants scenario

Reply:
cool, sounds good to me.